opentelekomcloud/resource_opentelekomcloud_cce_node_pool_v3.go

```diff
@@ -0,0 +1,526 @@
+			},
+			Type: d.Get("type").(string),
+		},
+	}
+
+	clusterid := d.Get("cluster_id").(string)
+	_, err = nodepools.Update(nodePoolClient, clusterid, d.Id(), updateOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error updating Open Telekom Cloud Node Pool: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Synchronizing"},
+		Target:     []string{""},
+		Refresh:    waitForCceNodePoolActive(nodePoolClient, clusterid, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      15 * time.Second,
+		MinTimeout: 5 * time.Second,
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error creating Open Telekom Cloud CCE Node Pool: %s", err)
+	}
+
+	return resourceCCENodePoolRead(d, meta)
+}
+
+func resourceCCENodePoolDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	nodePoolClient, err := config.cceV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating Open Telekom Cloud CCE client: %s", err)
+	}
+	clusterid := d.Get("cluster_id").(string)
+	err = nodepools.Delete(nodePoolClient, clusterid, d.Id()).ExtractErr()
+	if err != nil {
+		return fmt.Errorf("Error deleting Open Telekom Cloud CCE Node Pool: %s", err)
+	}
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Deleting"},
+		Target:       []string{"Deleted"},
+		Refresh:      waitForCceNodePoolDelete(nodePoolClient, clusterid, d.Id()),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		Delay:        60 * time.Second,
+		PollInterval: 20 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting Open Telekom Cloud CCE Node Pool: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func waitForCceNodePoolActive(cceClient *golangsdk.ServiceClient, clusterId, nodePoolId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		n, err := nodepools.Get(cceClient, clusterId, nodePoolId).Extract()
+		if err != nil {
+			return nil, "", err
+		}
+		return n, n.Status.Phase, nil
+	}
+}
+
+func waitForCceNodePoolDelete(cceClient *golangsdk.ServiceClient, clusterId, nodePoolId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		log.Printf("[DEBUG] Attempting to delete Open Telekom Cloud CCE Node Pool %s.\n", nodePoolId)
+
+		r, err := nodepools.Get(cceClient, clusterId, nodePoolId).Extract()
+
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				log.Printf("[DEBUG] Successfully deleted Open Telekom Cloud CCE Node Pool %s", nodePoolId)
+				return r, "Deleted", nil
+			}
+			return r, "Deleting", err
+		}
+
+		log.Printf("[DEBUG] Open Telekom Cloud CCE Node Pool %s still available.\n", nodePoolId)
+		return r, r.Status.Phase, nil
+	}
+}
+
+func recursiveNodePoolCreate(cceClient *golangsdk.ServiceClient, opts nodepools.CreateOptsBuilder, ClusterID string, errCode int) (*nodepools.NodePool, string) {
```

Comment:
You can drop `errCode` argument (as its value is always the same) and use `error` instead of `string` in return values.